bat/ASM.BAT

```diff
@@ -11,20 +11,23 @@ if /I "%1" == "tron" (set z_TraceMode=tron
 set /A z_NestLevel=%z_NestLevel%+1
 rem ----- Lvl(%z_NestLevel%) Start %0 %1 %2 %3 %4 %5 %6 %7 %8 %9
 
-if /I %1. == . (set /P z_file=Hit Enter for help or suppply name of mlc file to assemble: 
+if /I %1. == . (set /P z_file=Hit Enter for help or supply name of mlc file to assemble: 
         ) else (set z_file=%1)
 if /I %z_file%. == . goto help
 :chkfile
 rem convert Unix path notation to Windows before checking the file
 set z_file=%z_file:/=\%
-if exist %z_file%.MLC goto file_ok
-rem if user specified valid extension, strip it off
-if /I %z_file:~-4%. NEQ .MLC. goto help
-set z_file=%z_file:~0,-4%
-if exist %z_file%.MLC goto file_ok
-echo %0 ERROR: %z_file%.MLC was not found
-set z_ReturnCode=16
-goto return
+call bat\fparts.bat %z_TraceMode% %z_file%
+set z_file=%z_fdrv%%z_fpath%%z_fname%
+if exist %z_file%%z_fext% goto file_ok
+rem if extension specified: error because no default applies
+if /I %z_fext%. NEQ . goto notfound
+rem try with default extension of mlc
+set z_fext=.mlc
```

Comment:
you are correct. I have changed mlc to uppercase wherever needed in bat files in the bat subdirectory. I have not ventured to other directories to avoid scope creep.